tracer/src/Datadog.Trace/DataStreamsMonitoring/Utils/BinaryPrimitivesHelper.cs

```diff
@@ -0,0 +1,76 @@
+﻿// <copyright file="BinaryPrimitivesHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+
+namespace Datadog.Trace.DataStreamsMonitoring.Utils;
+
+internal static class BinaryPrimitivesHelper
+{
+    public static void WriteUInt64LittleEndian(byte[] bytes, ulong value)
+    {
+#if NETCOREAPP3_1_OR_GREATER
+        System.Buffers.Binary.BinaryPrimitives.WriteUInt64LittleEndian(bytes, value);
+#else
+        // write the bytes one at a time
+        if (BitConverter.IsLittleEndian)
+        {
+            bytes[0] = unchecked((byte)(value & 0xFF));
+            bytes[1] = unchecked((byte)((value >> 8) & 0xFF));
+            bytes[2] = unchecked((byte)((value >> 16) & 0xFF));
+            bytes[3] = unchecked((byte)((value >> 24) & 0xFF));
+            bytes[4] = unchecked((byte)((value >> 32) & 0xFF));
+            bytes[5] = unchecked((byte)((value >> 40) & 0xFF));
+            bytes[6] = unchecked((byte)((value >> 48) & 0xFF));
+            bytes[7] = unchecked((byte)((value >> 56) & 0xFF));
+        }
+        else
+        {
+            bytes[7] = unchecked((byte)(value & 0xFF));
+            bytes[6] = unchecked((byte)((value >> 8) & 0xFF));
+            bytes[5] = unchecked((byte)((value >> 16) & 0xFF));
+            bytes[4] = unchecked((byte)((value >> 24) & 0xFF));
+            bytes[3] = unchecked((byte)((value >> 32) & 0xFF));
+            bytes[2] = unchecked((byte)((value >> 40) & 0xFF));
+            bytes[1] = unchecked((byte)((value >> 48) & 0xFF));
+            bytes[0] = unchecked((byte)((value >> 56) & 0xFF));
+        }
+
+#endif
+    }
+
+    public static ulong ReadUInt64LittleEndian(byte[] bytes)
+    {
+#if NETCOREAPP3_1_OR_GREATER
+        return System.Buffers.Binary.BinaryPrimitives.ReadUInt64LittleEndian(bytes);
+#else
+        // read the bytes one at a time
+        if (BitConverter.IsLittleEndian)
+        {
+            ulong value = bytes[7];
```

Comment:
Should we validate the size of the byte array just for extra precaution on the usage? And do the same in the Write method

Reply:
Sure, though in that case I'm going to throw an exception exactly as if I didn't check it, so don't know if it's worth it? 😄

Reply:
It's one of the rare cases where I think checking arguments is worth it personally as it is not clear in the prototype that you expect an array of a specific size. But I don't mind much if you don't add the check :)

Reply:
done https://github.com/DataDog/dd-trace-dotnet/pull/3131/commits/efb98a819f30c6914586eace8afb147cd42a89a2